src/context.rs

```diff
@@ -1237,52 +1266,229 @@ impl SeafowlContext for DefaultSeafowlContext {
+                                                project_filter,
+                                                &[expr.clone()],
+                                                self.internal_object_store.inner.clone(),
+                                            )
+                                            .await?;
+
+                                        let projection_plan =
+                                            Arc::new(ProjectionExec::try_new(
+                                                projection_exprs.clone(),
+                                                projection_input_plan,
+                                            )?);
+
+                                        // Filter out the old rows that are updated via the projection plan
+                                        let delete_plan = table
+                                            .partition_filter_plan(
+                                                group,
+                                                delete_filter,
+                                                &[expr.clone().not()],
+                                                self.internal_object_store.inner.clone(),
+                                            )
+                                            .await?;
+
+                                        // Merge the two plan outputs for the final update plan
+                                        update_plan = Arc::new(UnionExec::new(vec![
+                                            projection_plan,
+                                            delete_plan,
+                                        ]));
+                                    }
+                                    (_, _, _) => {
+                                        // No selection was specified, update all rows in the partitions
+                                        let scan_plan = table
+                                            .partition_scan_plan(
+                                                &None,
+                                                group,
+                                                &[],
+                                                None,
+                                                self.internal_object_store.inner.clone(),
+                                            )
+                                            .await?;
+
+                                        update_plan = Arc::new(ProjectionExec::try_new(
+                                            projection_exprs.clone(),
+                                            scan_plan,
+                                        )?);
+
+                                        println!("table schema: {:?}", table.schema());
+                                        println!(
+                                            "update schema: {:?}",
+                                            update_plan.schema()
+                                        );
```

Comment:
```suggestion
                                        debug!("table schema: {:?}", table.schema());
                                        debug!(
                                            "update schema: {:?}",
                                            update_plan.schema()
                                        );
```

Reply:
Thanks, I'll rather remove this completely, this was just for debugging.